src/main/java/edu/harvard/iq/dataverse/DataverseFacetServiceBean.java

```diff
@@ -42,23 +42,24 @@ public void delete(DataverseFacet dataverseFacet) {
         cache.invalidate();
     }
     
-	public void deleteFacetsFor( Dataverse d ) {
-		em.createNamedQuery("DataverseFacet.removeByOwnerId")
-			.setParameter("ownerId", d.getId())
-				.executeUpdate();
+    public void deleteFacetsFor(Dataverse d) {
+        em.createNamedQuery("DataverseFacet.removeByOwnerId")
+                .setParameter("ownerId", d.getId())
+                .executeUpdate();
         cache.invalidate(d.getId());
-        
-	}
-	
+
+    }
+
     public DataverseFacet create(int displayOrder, DatasetFieldType fieldType, Dataverse ownerDv) {
         DataverseFacet dataverseFacet = new DataverseFacet();
-        
+
         dataverseFacet.setDisplayOrder(displayOrder);
         dataverseFacet.setDatasetFieldType(fieldType);
         dataverseFacet.setDataverse(ownerDv);
-        
-        ownerDv.getDataverseFacets().add(dataverseFacet);
+
         em.persist(dataverseFacet);
+        ownerDv.getDataverseFacets().add(dataverseFacet);
```

Comment:
Just curious - did this have to move after the persist?

Reply:
maybe not. I put it back and it still seems to work